### PR TITLE
Move top command to it's own package

### PR DESCRIPTION
**Pull request by shiywang**
just clean up, no feature changed
/assign @alexandercampbell 
/assign @mengqiy 
Move kubectl top command to it's own package

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/top/top_pod_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cmd
+package top
```
Comment (alexandercampbell) [inline pkg/kubectl/cmd/top/top_pod_test.go]: If these files are now in a `top` directory, you should be able to remove the `top_` prefix on the filenames.

**Review by alexandercampbell — COMMENTED** [review]
I'm okay with this. My only question is: why?

**Comment by mengqiy** [comment]
This PR doesn't need a release note.
Please move the description to the PR body and please also include what subcommands it has in the PR body as well.

File: pkg/kubectl/cmd/top/top_pod_test.go (hunk 1 above)
Comment (shiywang) [inline pkg/kubectl/cmd/top/top_pod_test.go]: @mengqiy I checked that `set` package also has `set_` prefix, do you think that also should be removed as well ?

Comment (mengqiy) [inline pkg/kubectl/cmd/top/top_pod_test.go]: I agree with what @alexandercampbell says.
It will be good if someone can fix the naming for `set`.

**Comment by alexandercampbell** [comment]
@shiywang introducing more packages complicates the dependency graph and I'm curious what benefit we receive. What was the reasoning behind this change? Thanks!

**Pull request closed without merging**